app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingFragmentTest.kt

```diff
@@ -480,21 +489,52 @@ class OnboardingFragmentTest {
   }
 
   @Test
-  // TODO(#973): Fix OnboardingFragmentTest
-  @Ignore
   fun testOnboardingFragment_clickOnSkip_changeOrientation_titleIsCorrect() {
     launch(OnboardingActivity::class.java).use {
       onView(withId(R.id.skip_text_view)).perform(click())
+      testCoroutineDispatchers.runCurrent()
       onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
       onView(
         allOf(
           withId(R.id.slide_title_text_view),
-          isDisplayed()
+          isCompletelyDisplayed()
         )
       ).check(matches(withText(R.string.onboarding_slide_3_title)))
     }
   }
 
+  private fun swipeToSlide1() {
+    onView(withId(R.id.onboarding_slide_view_pager)).perform(swipeLeft())
+    testCoroutineDispatchers.runCurrent()
+  }
+
+  private fun swipeToSlide2() {
+    swipeToSlide1()
+    swipeToSlide1()
+  }
+
+  private fun swipeToSlide3() {
+    swipeToSlide2()
+    swipeToSlide1()
+  }
+
+  private fun childAtPosition(parentMatcher: Matcher<View?>, childPosition: Int): Matcher<View?> {
+    return object : TypeSafeMatcher<View?>() {
+      override fun describeTo(description: Description) {
+        description.appendText("with $childPosition child view of type parentMatcher")
+      }
+
+      override fun matchesSafely(view: View?): Boolean {
+        if (view?.parent !is ViewGroup) {
+          return parentMatcher.matches(view?.parent)
+        }
+        val group = view.parent as ViewGroup
+        return parentMatcher.matches(view.parent) && group.getChildAt(childPosition) == view
+      }
+    }
+  }
+
```

Comment:
@BenHenning I tried implementing a custom ViewMatcher, but I couldn't get rid of `allOf`.